### DISCUSSION TITLE
Implement Bazel module lockfile support

### DISCUSSION THREAD

**Discussion by cgrindel** [renovatebot/renovate#25532] [Suggest an Idea]
### Requirements

- Read the `.bazelversion`. Ensure that it is 6.4.0 or later.
- Read the bazelrc files looking for `--lockfile_mode`. If not found, it is assumed to be `update`.
- Lockfile Mode
  - `update`: update the lockfile. 
  - `error`: do not update the lockfile. NOTE: It is unclear why one would configure Renovate to update Bazel and have their lockfile mode be `error`.
  - `off`: do not update the lockfile.
- If a lockfile update is warranted, execute `bazel mod deps`.
- Need to ensure that [bazelisk](https://github.com/bazelbuild/bazelisk/) is installed on the platforms that execute Renovate (e.g., Docker images).

### Related Links

- [Bazel module lockfile documentation](https://bazel.build/external/lockfile)
- [Lockfile maintenance config](https://github.com/renovatebot/renovate/blob/5d9296e0b9ce5dbd503871f7d9ca5a8cca4eab83/docs/usage/configuration-options.md#lockfilemaintenance)
- [Lockfile maintenance schedule](https://github.com/renovatebot/renovate/blob/837c8850d848e32c056a0c42e9c1a756a6e7b4eb/docs/usage/faq.md#keep-lock-files-including-sub-dependencies-up-to-date-even-when-packagejson-hasnt-changed)

**Comment by cgrindel** [comment]
> Need to ensure that [bazelisk](https://github.com/bazelbuild/bazelisk/) is installed on the platforms that execute Renovate (e.g., Docker images).

If I have that point right, besides the Docker images provided by Renovate, are there others that we need to update?

**Comment by rarkins** [comment]
Created issue: #25557
